libosmscout-import/src/osmscout/import/GenLocationIndex.cpp

```diff
@@ -466,35 +610,59 @@ namespace osmscout {
     return true;
   }
 
-  void LocationIndexGenerator::AddRegion(Region& parent,
-                                         const RegionRef& region)
+  bool LocationIndexGenerator::AddRegion(Region& parent,
+                                         RegionRef& region,
+					 bool assume_contains)
   {
+    bool added = false;
     for (const auto& childRegion : parent.regions) {
-      if (region->CouldContain(*childRegion)) {
-        for (const auto& regionArea : region->areas) {
-          for (const auto& childRegionArea : childRegion->areas) {
-            if (IsAreaSubOfAreaOrSame(regionArea,childRegionArea)) {
-              // If we already have the same name and are a "minor" reference, we skip...
-              if (!(region->name==childRegion->name &&
-                    region->reference.type<childRegion->reference.type)) {
-                AddRegion(*childRegion,region);
-              }
-
-              return;
-            }
-          }
-        }
+      if ( childRegion->CouldContain(*region, true) ) {
+	added = AddRegion(*childRegion,region,false);
+	if (added) return true;
```

Comment:
Yes, but for that `SortInBoundaries` calls `AddRegion` with the default arguments (i.e. `assume_contains=true` in this case)